### ISSUE TITLE
Recommend disabling -Werror in the example repository build files

### ISSUE THREAD

**Issue by phillipjohnston** [johnttaylor/pim#4]
The -Werror flag works fine when build environments are precisely controlled. However, this public code is not presented with a precisely controlled environment, leading to a variation in possible compiler versions that will be used with it. -Werror is problematic because:

- As newer compilers are released, new warnings will eventually be generated due to improvements on the compiler side (even with no changes to the code). Then the build won't complete successfully.
- If someone uses an older/different compiler that does not support a particular warning flag, but would otherwise build successfully, -Werror will cause the build to fail. For example, clang does not use [these particular warning flags](https://github.com/johnttaylor/pim/blob/9faf16725d4c682d91f4f41a83e69ef3fbe67ce3/xsrc/nqbp/nqbplib/toolchains/linux/gcc/console_exe.py#L26), but whether or not these warnings are successfully disabled does not impact someone's ability to try out the project.
- It's also the case that Clang can be used to build this project, but it is likely to generate different warnings (such as reported in #3). -Werror stops the project from being built, even though the project simulator builds fine with clang.

Which means that users facing these challenges are going to have to turn off -Werror anyway to play around with the project.

**Comment by johnttaylor** [comment]
I agree with your comments newer/different/older compilers causing problems with respect to compile time options (especially warning options).  Do better handle/deal with different compiler version I will be providing a fix mentioned in #6 

TL;DR - if want to turn off the -Wall -Werror flags - simply edit the the mytoolchain.py file(s) in your build directory and remove the -Wall/-Werror flags from the line setting the variable: `base_release.cflags` (e.g. change to: `base_release.cflags    = '-m32 -std=c++11 -x c++ -fprofile-arcs -ftest-coverage -DCATCH_CONFIG_FAST_COMPILE'`

The longer version....
The problems arising from using the Clang compiler is a build script issue, i.e. you are trying to use a build script that is specific to the GCC compiler with the Clang compiler (see rule 3 "Similar but different, is still different" ;-).  I don't currently have the Clang compiler installed on any my systems so I can't give any timeline for when I can supply a Clang compiler specific build script.  However, below is description of the changes that would need to be made.  If you are willing to make the changes and verify the build script, I will incorporate your changes into the repo.

Note: The instructions are probably not 100% correct because I am not able to reproduce the issues on my systems.  If you have any questions or need help please don't hesitant to ping-me.  Also, reading Chapter 16: About NQBP will provide a general overview on how the NQBP engine works (unfortunately I don't have any documentation for how to update/maintain/extended NQBP :( )

1. In the `xsrc/nqbp/nqbplib/toolchains/linux/` directory, create a new sub-directory: `clang` and copy the contents of `xsrc/nqbp/nqbplib/toolchains/gcc/` into the new directory
2. In the new `xsrc/nqbp/nqbplib/toolchains/clang/console_exe.py` file - make the following changes (per your feedback #9)
```
class ToolChain( base.ToolChain ):

    #--------------------------------------------------------------------------
    def __init__( self, exename, prjdir, build_variants, default_variant='release' ):
        base.ToolChain.__init__( self, exename, prjdir, build_variants, default_variant )
        self._ccname = 'Clang'
        self._linker_libgroup_start = ""
        self._linker_libgroup_end   = ""

        #
        # Build Config/Variant: "xyz"
        #
       
        # Disable the following 'string warnings' because they generate false-positives (i.e. issues with GCC itself)
        self._base_release.cflags = self._base_release.cflags + '-Wno-stringop-truncation -Wno-stringop-overflow '
        self._base_release.linklibs  = '-Wl,-lc++ -Wl,-lm'

        #self._base_xyz = self._base_release.copy()
        #self._base_xyz.cflags  = '-c -DBUILD_TIME_UTC={:d}'.format(self._build_time_utc)
```
3. Then under every (unfortunately) project or test executable you want to build using the Clang compiler, create a new `linux/clang` directory (e.g. `tests/Cpl/System/_0test/realtime/linux/clang`)
4. Go to the 'peer' gcc directory (e.g. `tests/Cpl/System/_0test/realtime/linux/gcc`) and run `nqbp.py z` to remove all derived files.
5. Copy the content of the 'peer' gcc directory to the new `clang/` directory.
6. Modify line 152 in the new `mytoolchain.py` file in the new `clang/` to be: `from nqbplib.toolchains.linux.clang.console_exe import ToolChain`

**Comment by phillipjohnston** [comment]
The problems arising from using the Clang compiler is a build script issue, i.e. you are trying to use a build script that is specific to the GCC compiler with the Clang compiler (see rule 3 "Similar but different, is still different" ;-).

Sorry, I perhaps gave the wrong impression here - I understand that Clang is not supported and I had no expectation that it would be . At the time it was just faster for me to hack around with it than install and set up a new VM.

> However, below is description of the changes that would need to be made. If you are willing to make the changes and verify the build script, I will incorporate your changes into the repo.

Happy to tackle that.
